layers/README.md

```diff
@@ -0,0 +1,85 @@
+# layers
+
+This folder contains modules and scripts for working with ATT&CK layers. The core module allows users to load, validate, process, and save ATT&CK layers. A brief overview of the components can be found below. All scripts adhere to the MITRE ATT&CK Navigator Layer file format, [version 3.0](https://github.com/mitre-attack/attack-navigator/blob/develop/layers/LAYERFORMATv3.md).
+
+| script | description |
+|:-------|:------------|
+| [exceptions.py](core/exceptions.py) | Implements errors and supporting functions for the core module. |
+| [filter.py](core/filter.py) | Implements a basic filter object. |
+| [gradient.py](core/gradient.py) | Implements a basic gradient object. |
+| [layer.py](core/layer.py) | Provides an interface for interacting with core module's layer representation. A further breakdown can be found in the corresponding section below. |
+| [layerobj.py](core/layerobj.py) | Implements a basic layer object. This should not be manipulated directly, instead, use the [layer representation](core/layer.py). |
+| [layout.py](core/layout.py) | Implements a basic layout object. |
+| [legenditem.py](core/legenditem.py) | Implements a basic legenditem object. |
+| [metadata.py](core/metadata.py) | Implements a basic metadata object. |
+| [technique.py](core/technique.py) | Implements a basic technique object. |
+| [layerops.py](manipulators/layerops.py) | Provides a means by which to combine multiple ATT&CK layer objects in customized ways. A further breakdown can be found in the corresponding section below. |
+
+## layer.py
+Layer.py provides a Layer class, which is the interface used to interact with the rest of the core module, and provides an abstract representation of an ATT&CK layer. The layer class has a collection of methods, the functionality of which is documented in the table below. An example of how to interface with the class to load and retrieve layer objects is documented following that.
+
+| method | description |
+|:-------|:------------|
+| Layer().load_input(_input_) | Loads an ATT&CK layer from either a dictionary or a string representation of a dictionary. |
+| Layer().load_file(_input_) | Loads an ATT&CK layer from a file location specified by the _input_. |
+| Layer().export_file(_input_) | Saves the current state of the loaded ATT&CK layer to a json file denoted by the _input_. |
+| Layer().get_dict() | Returns a representation of the current ATT&CK layer object as a dictionary. | 
+
+#### Example Usage
+
+```python
+example_layer_dict = {
+    "name": "example layer",
+    "version": "3.0",
+    "domain": "mitre-enterprise",
+    "description": "hello, world",
+    ...
+    }
```

Comment:
You should include an actual entire layer here. Putting "..." in a usage example isn't a good idea because users may want to copy and paste the example into a script as a starting point.

Reply:
I'm really hesitant to do that given that layer files are kinda large, but okay, we can do that.

Reply:
A layer with the minimum required set of fields shouldn't be too huge I think.